### PR TITLE
Enable erase_chip by default

### PR DESCRIPTION
**Pull request by haslinghuis**
When migrating from previous version this settings was reported to disabled by default.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/tabs/firmware_flasher.js
Comment (McGiverGim) [inline src/js/tabs/firmware_flasher.js:582]: I think is better to "overcharge" the getConfig() method with this code, adding a second "optional" parameter as "default". If configured, and undefined, then return the default. In this way we can do the same for others parameters if we want. What do you think?

Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js:582]: Agree, its a better option, but then we have to change them all.

Comment (McGiverGim) [reply src/js/tabs/firmware_flasher.js]: Seems good to me 😉 thanks!

File: src/js/tabs/firmware_flasher.js
Comment (haslinghuis) [inline src/js/tabs/firmware_flasher.js:581]: here it is disabled by default if key is not present or false

Comment (nerdCopter) [reply src/js/tabs/firmware_flasher.js:581]: everything "fixed" by only setting this `true` always; nothing more needed.
and thusly, pilots can disable manually per configurator execution.
it will automatically re-enable upon re-launch.

no reason this should be saved locally. otherwise, pilots will break their own quads repeatedly.

i did this in "other project" in 2020, no complaintive feedback by pilots.

Comment (nerdCopter) [reply src/js/tabs/firmware_flasher.js:581]: not done cleanly, but this: https://github.com/emuflight/EmuConfigurator/pull/180/files

Comment (haslinghuis) [reply src/js/tabs/firmware_flasher.js:581]: Yes this PR has same outcome as in yours it was better to remove the condition as both paths leads to true but we don't need to nuke the setting on startup :)

**Comment by ctzsnooze** [comment]
Can I confirm the expected behaviour after this PR...
- if `Full Chip Erase` was active when entering the Flashing tab, nothing changes.
- if `Full Chip Erase` was not active when the Flashing tab was opened, it is changed to active, every time we enter the Flashing Tab?
- if `Full Chip Erase` was not active when the Flashing tab was opened, and was automatically changed to active, the user can only only be reverted it to `not active` by clicking the switch to `off`?
- If the user has set `Full Chip Erase` to off, and exits the Flashing tab, does something else in some other tab, and then returns to the Flashing tab, it will remain off? 
- - If the user has set `Full Chip Erase` to off, and quits Configurator, then re-opens and goes to the Flashing tab, it will show `Full Chip Erase` as being `on`?

**Comment by haslinghuis** [comment]
> In the fourth possibility, that a user turns Full Chip Erase off, goes to some other tab, and comes back to the Flashing tab, I guess it is best for Full Chip Erase to stay off.

That behavior was covered by the first commit.

**Comment by nerdCopter** [comment]
tested `4c5a914f`:
- 1 = yes, remains active
- 2 = yes, reverts to active
- 3 = yes, can only manually disable
- 4 = no, returns to enabled
- 5 = yes, reverts enabled

further findings. seems flashing significantly disparate firmware version, it will still erase chip even if disabled.
i've seen this in the far past and did not comment such. 
for example when disabling erase, flashing master to master works as expect, master to PR 13248 will work as expected, but some older PR like 13234 will full-chip erase.

**Review by nerdCopter — APPROVED** [review]

**Comment by haslinghuis** [comment]
To cover point 4 - setting default enabled value in main.js instead of in firmware flasher tab would allow for session wide behavior.

**Comment by nerdCopter** [comment]
> To cover point 4 - setting default enabled value in main.js instead of in firmware flasher tab would allow for session wide behavior.

i am okay with current behavior. unsure what others desire.

**Review by blckmn — APPROVED** [review]

**Pull request merged**
